databuilder/query_engines/mssql.py

```diff
@@ -63,11 +66,42 @@ def get_query(self, variable_definitions):
         results_query = super().get_query(variable_definitions)
         # Write results to a temporary table and select them from there. This allows us
         # to use more efficient/robust mechanisms to retrieve the results.
+        table_name, schema = self.get_results_table_name_and_schema(
+            self.config.get("TEMP_DATABASE_NAME")
+        )
         results_table = temporary_table_from_query(
-            "#results", results_query, index_col="patient_id"
+            table_name, results_query, index_col="patient_id", schema=schema
         )
         return sqlalchemy.select(results_table)
 
+    def get_results_table_name_and_schema(self, temp_database_name):
+        # If we have a temporary database we can write results their which enables
+        # us to continue retrieving results after an interrupted connection
+        if temp_database_name:
+            # As the table is not session-scoped it needs a unique name
+            timestamp = datetime.datetime.utcnow()
+            token = secrets.token_hex(6)
+            table_name = f"results_{timestamp:%Y%m%d_%H%M}_{token}"
+            # The `schema` variable below is actually a multi-part identifier of the
+            # form `<database-name>.<schema>`. We don't really care about the schema
+            # here, we just want to use whatever is the default schema for the database.
+            # MSSQL allows you to do this by specifying "." as the schema name. However
+            # I can't find a way of supplying this without SQLAlchemy's quoting
+            # algorithm either mangling it or blowing up. We could work around this by
+            # attaching our own Identifier Preparer to our MSSQL dialect, but that
+            # sounds a lot like hard work. So we're using the default, default schema of
```

Comment:
```suggestion
            # sounds a lot like hard work. So we're using the default schema of
```

Reply:
Ah, hilariously, this isn't actually a typo: I really do mean two levels of default. When you create databases in MSSQL they have a default schema set. The default value for this default schema is `dbo`. But it's possible to create databases with a different default schema. I could probably word this a bit more clearly though!